File: app/src/main/java/com/github/dawidhyzy/kotlinvsjava/jv/forecast/ForecastContract.java
package com.github.dawidhyzy.kotlinvsjava.jv.forecast;

import com.github.dawidhyzy.kotlinvsjava.jv.domain.Forecast;

/**
 * @author Dawid Hyży <dawid.hyzy@seedlabs.io>
 * @since 31/01/16.
 */
public class ForecastContract {

    public interface View{
        void showLoading(boolean show);
        void showError(String message);
        void setForecast(Forecast forecast);
    }

    public interface Presenter{
        void loadForecast();
        void refresh();
    }
}

File: app/src/main/java/com/github/dawidhyzy/kotlinvsjava/kt/forecast/ForecastContract.kt
package com.github.dawidhyzy.kotlinvsjava.kt.forecast

import com.github.dawidhyzy.kotlinvsjava.kt.domain.Forecast

/**
 * @author Dawid Hyży <dawid.hyzy@seedlabs.io>
 * @since 06/02/16.
 */
interface View {
    fun showLoading(show: Boolean)
    fun showError(message: String)
    fun setForecast(forecast: Forecast)
}

interface Presenter {
    fun loadForecast()
    fun refresh()
}